--- cantata-0.8.2/devices/device.h	2012-08-01 19:49:57.896777183 -0500
+++ cantata-0.8.2.new/devices/device.h	2012-08-01 19:53:40.890979441 -0500
@@ -29,7 +29,9 @@
 #include "encoders.h"
 #include "deviceoptions.h"
 #include <QtCore/QObject>
+#ifdef ENABLE_KDE_SUPPORT
 #include <solid/device.h>
+#endif
 
 class QWidget;
 class DevicesModel;
@@ -64,6 +66,7 @@
         RemoteFs
     };
 
+#ifdef ENABLE_KDE_SUPPORT
     Device(DevicesModel *m, Solid::Device &dev)
         : MusicLibraryItemRoot(dev.vendor()+QChar(' ')+dev.product())
         , model(m)
@@ -73,6 +76,7 @@
         , needToFixVa(false)
         , jobAbortRequested(false) {
     }
+#endif
     Device(DevicesModel *m, const QString &name)
         : MusicLibraryItemRoot(name)
         , model(m)
@@ -84,9 +88,11 @@
     virtual ~Device() {
     }
 
+#ifdef ENABLE_KDE_SUPPORT
     virtual QString icon() const {
         return solidDev.isValid() ? solidDev.icon() : QLatin1String("folder");
     }
+#endif
     virtual QString coverFile() const {
         return QString();
     }
@@ -110,9 +116,11 @@
     virtual void removeCache() {
     }
 
+#ifdef ENABLE_KDE_SUPPORT
     virtual QString udi() const {
         return solidDev.udi();
     }
+#endif
     void applyUpdate();
     bool haveUpdate() const {
         return 0!=update;
@@ -176,7 +184,9 @@
     DevicesModel *model;
     DeviceOptions opts;
     bool configured;
+#ifdef ENABLE_KDE_SUPPORT
     Solid::Device solidDev;
+#endif
     MusicLibraryItemRoot *update;
     Song currentSong;
     QString currentBaseDir;
