--- a/kdiff3/src-QT4/kdiff3.cpp
+++ b/kdiff3/src-QT4/kdiff3.cpp
@@ -211,6 +211,8 @@
       m_outputFilename = args->getOption("output");
       if ( m_outputFilename.isEmpty() )
          m_outputFilename = args->getOption("out");
+      if ( ! m_outputFilename.isEmpty() )
+         m_outputFilename = FileAccess( m_outputFilename, true ).absoluteFilePath();
    }
 
    m_bAutoFlag = args!=0  && args->isSet("auto");
